### PR TITLE
Rename commands to fit a more consistent naming scheme

### PR DESCRIPTION
**Pull request by lenormf**
Hi,

As discussed in #806, the naming scheme of the commands can feel a bit inconsistent, so I decided to make my proposal for new names.

This PR does not modify anything other than the hardcoded names in `commands.cc`, as I think we should all agree (and by all I mean @mawww) on what names to use before changing the scripts.

Discuss.

### CODE REVIEW

File: src/commands.cc [hunk 1]
```diff
@@ -982,9 +982,9 @@ void define_command(const ParametersParser& parser, Context& context, const Shel
 }
 
 const CommandDesc define_command_cmd = {
+    "define",
```
Comment (mawww) [inline src/commands.cc]: shall we use a more explicit name ? `command` ?

File: src/commands.cc [hunk 2]
```diff
@@ -1251,9 +1251,9 @@ const CommandDesc unset_option_cmd = {
 };
 
 const CommandDesc declare_option_cmd = {
+    "declare",
```
Comment (mawww) [inline src/commands.cc]: A more explicit name ? `option` ?

File: src/commands.cc [hunk 3]
```diff
@@ -667,9 +667,9 @@ Highlighter& get_highlighter(const Context& context, StringView path)
 }
 
 const CommandDesc add_highlighter_cmd = {
+    "add-highlighter",
```
Comment (mawww) [inline src/commands.cc]: or just `highlight` ?

**Comment by lenormf** [comment]
>I wonder the kind of commands we need, change-buffer vs buffer, hook or add-hook...

That's the issue I hit when changing the names, and I've decided not to go completely for one naming format, even if that made the changeset inconsistent.

On one hand you have commands whose name is just a verb (`alias`, `map` etc), those are easy to remember and the name of their removal counter part simply starts with "un" (`unalias`, `unmap` etc). I could have made it so they all followed the `action-object` name format, but it seemed rather unnecessary and cumbersome (`add-alias`, `add-map` etc).

But on the other hand there are commands whose removal counter part would not sound so great if they followed the `verb` only convention: `unhook`, `unhighlight`… So since the name for the removing commands had to be prefixed with `remove-`, I made the regular command start with `add-`. That's why we don't have `highlight`/`unhighlight` or `hook`/`unhook`, but `add-highlighter`/`remove-highlighter` and `add-hook`/`remove-hooks` (the latter should be renamed to `remove-hook` by the way, even though the current command name is `rmhooks`).

The corner case of all this is `buffer`, I hesitated before changing its name because the entire world is so used to that command. But since this PR is based on the remark that it's not very complicated to intuitively find the correct command name with fuzzy matching, I think it's the right decision to steer away from the `vim` model.

File: src/commands.cc (hunk 2 above)
Comment (lenormf) [inline src/commands.cc]: I think it's better if we keep this command name a verb, in the name of consistency.

File: src/commands.cc (hunk 3 above)
Comment (lenormf) [inline src/commands.cc]: C.f. [my answer](https://github.com/mawww/kakoune/pull/827#issuecomment-251585573).

**Comment by Delapouite** [comment]
I also prefer a consistent naming scheme over vim old names.

**Comment by lenormf** [comment]
Should this be closed now that new names have been merged? Should we move on to option names?

**Comment by mawww** [comment]
Names have been changed through some other commits, thanks for the inspiration 👍

**Pull request closed without merging**